### PR TITLE
FIX: loadpkl failed when pklz file contained versioning info

### PR DESCRIPTION
**Pull request by stilley2**
## Summary

loadpkl called pickle.load at file position 0, which would fail if the first line of the file was json  versioning info (as when using savepkl with versioning=True). This was fixed by only seeking to position 0 if no versioning information is found.

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+def test_pickle(tmp_path):
+    testobj = 'iamateststr'
+    pickle_fname = str(tmp_path / 'testpickle.pklz')
+    utils.filemanip.savepkl(pickle_fname, testobj)
+    outobj = utils.filemanip.loadpkl(pickle_fname)
+    assert outobj == testobj
+
+
+def test_pickle_versioning(tmp_path):
```
Comment (effigies) [inline nipype/tests/test_utils.py]: What about parameterizing this into a single test?

```Python
@pytest.mark.parametrize("save_ver", (True, False))
@pytest.mark.parametrize("load_ver", (True, False))
def test_pickle(tmp_path, save_ver, load_ver):
    testobj = 'iamateststr'
    pickle_fname = str(tmp_path / 'testpickle.pklz')
    utils.filemanip.savepkl(pickle_fname, testobj, versioning=save_ver)
    outobj = utils.filemanip.loadpkl(pickle_fname, versioning=load_ver)
    assert outobj == testobj
```

**Comment by effigies** [comment]
@oesteban I think this issue was introduced in the previous version (#2985). I would recommend this to go into the 1.2.2 release, if it's ready in time.

File: nipype/tests/test_utils.py (hunk 1 above)
Comment (stilley2) [reply nipype/tests/test_utils.py]: Good idea.

**Comment by stilley2** [comment]
I actually just removed the versioning argument from loadpkl. All it was doing was altering the error reporting, and it seemed unnecessary (although maybe it was there for a reason?).

**Comment by effigies** [comment]
Looks like it was added in #2626, and what it seems to be doing is guarding against accessing the undefined `pkl_metadata` in the next couple lines. I would leave it for now, just because there doesn't seem to be a big gain in removing these lines, and there will be a semantic change.

We can open an issue to review that, though, if you think it will help with clarity.

**Comment by stilley2** [comment]
Ok I reverted it. However, I don't think it guards against that any more. I'll open up an issue.

File: nipype/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from nipype import utils
+import pytest
+
+
+@pytest.mark.parametrize("load_versioning", [True, False])
+@pytest.mark.parametrize("save_versioning", [True, False])
+def test_pickle(tmp_path, save_versioning, load_versioning):
```
Comment (effigies) [inline nipype/tests/test_utils.py]: Oh, this test should go in `nipype/utils/tests/test_filemanip.py`. Sorry, I just caught this.

Comment (stilley2) [reply nipype/tests/test_utils.py]: Oops I didn't see that file.

**Pull request merged**